detector/gpg.go

```diff
@@ -12,17 +12,17 @@ import (
 )
 
 // WatchGPG watches for hints that YubiKey is maybe waiting for a touch on a GPG request
-func WatchGPG(gpgPubringPath string, requestGPGCheck chan bool) {
-	// No need for a buffered channel,
-	// we are interested only in the first event, it's ok to skip all subsequent ones
-	events := make(chan notify.EventInfo)
+func WatchGPG(filesToWatch []string, requestGPGCheck chan bool) {
+	events := make(chan notify.EventInfo, len(filesToWatch))
```

Comment:
It's not necessary to add the second parameter here, the comment above is still valid even if we watch multiple files. In the end, we just want to get an event, a signal to tell us to perform an expensive check operation - having one signal is more than enough, it won't really help (and maybe even hurt) if we allow to buffer 5 signals, then having to perform the expensive check 5 times in a row.

Without the second parameter, we stop receiving file events until we execute the expensive operation to check whether yubikey is waiting for a touch, but it's okay, that expensive check will detect the touch regardless of which file originally triggered the event.

So I think let's return the old code snippet here :)

Reply:
Good point. We return to the old code snippet.